src/com/facebook/buck/jvm/java/PluginLoaderJavaFileManager.java

```diff
@@ -0,0 +1,36 @@
+package com.facebook.buck.jvm.java;
+
+import com.google.common.collect.ImmutableList;
+import javax.tools.StandardJavaFileManager;
+import javax.tools.StandardLocation;
+
+public class PluginLoaderJavaFileManager extends ForwardingStandardJavaFileManager
+    implements StandardJavaFileManager {
+
+  private final PluginFactory pluginFactory;
+  private final ImmutableList<JavacPluginJsr199Fields> javacPlugins;
+
+  /**
+   * Creates a new instance of ForwardingJavaFileManager.
+   *
+   * @param fileManager delegate to this file manager
+   */
+  protected PluginLoaderJavaFileManager(
+      StandardJavaFileManager fileManager,
+      PluginFactory pluginFactory,
+      ImmutableList<JavacPluginJsr199Fields> javacPlugins) {
+    super(fileManager);
+    this.pluginFactory = pluginFactory;
+    this.javacPlugins = javacPlugins;
+  }
+
+  @Override
+  public ClassLoader getClassLoader(Location location) {
+    // We only provide the shared classloader if there are plugins defined.
+    if (StandardLocation.ANNOTATION_PROCESSOR_PATH.equals(location) &&
```

Comment:
Under what circumstances is `getClassLoader` called? What happens when we have both annotation processors and plugins? We instantiate all APs in batch and call `CompilationTask::setProcessors` once in `Jsr199JavaInvocation`, but I'm wondering if there might be weird cases where we end up using the wrong classloader for an AP somehow.

Reply:
On my local tests no AP was interfered by this and I could find also no indicators on javac (searching on open JDK code) that loaded processors again even if you provide loaded ones. So in my view, as we do the `setProcessors` call to set the processors, always, both can coexist fine.